utils.go

```diff
@@ -161,6 +161,57 @@ func parseBinaryDateTime(num uint64, data []byte, loc *time.Location) (driver.Va
 	return nil, fmt.Errorf("invalid DATETIME packet length %d", num)
 }
 
+func formatDateTime(t time.Time) (buf [32]byte, n int, err error) {
```

Comment:
How about returning a slice, instead of array + n?  Is it increase allocation?

Reply:
Even better: take `buf []byte` as input.

Reply:
Already used `buf []byte`